staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go

```diff
@@ -336,14 +344,43 @@ func toolingBaseName(toolAnnotation string) string {
 	return toolAnnotation
 }
 
-func parseResourcesAnnotation(annotations map[string]string, mapper meta.RESTMapper) (map[schema.GroupVersionResource]*meta.RESTMapping, error) {
-	annotation, ok := annotations[ApplySetGRsAnnotation]
+// kindInfo holds type information about a particular resource type.
+type kindInfo struct {
+	restMapping *meta.RESTMapping
+}
+
+func parseKindAnnotation(annotations map[string]string, mapper meta.RESTMapper) (map[schema.GroupKind]*kindInfo, error) {
+	annotation, ok := annotations[ApplySetGKsAnnotation]
 	if !ok {
+		if annotations[DeprecatedApplySetGRsAnnotation] != "" {
```

Comment:
Given this functionality is in alpha / feature-flagged, I don't know if we want to be compatible with the existing applysets.  Three obvious options:
1) Do this and drop it after a few releases
2) Implement `fsck` that can map from the old annotation to the new one (we proposed this tool in our KEP, it will be handy anyway)
3) Just switch to the new annotation, it's alpha after all.